packages/component-test-utils-react/src/dispatcher/index.js

```diff
@@ -35,18 +34,35 @@ class Dispatcher {
     const hookIndex = this._getHookIndex();
 
     if (this._firstCall) {
-      this.states.push(initialState);
+      this._hookMemo.push(initialState);
     }
 
     return [
-      this.states[hookIndex],
+      this._hookMemo[hookIndex],
       newValue => {
-        this.states[hookIndex] = newValue;
+        this._hookMemo[hookIndex] = newValue;
         // Updating the state trigger a render
         this._shallowedComponent._render();
       }
     ];
   }
+
+  useEffect(fn, memo) {
+    const hookIndex = this._getHookIndex();
+
+    const isSameMemo =
+      this._hookMemo[hookIndex] &&
+      (this._hookMemo[hookIndex] === memo ||
+        this._hookMemo[hookIndex].reduce(
+          (isSameMemo, prop, i) => isSameMemo && memo[i] === prop,
+          true
+        ));
```

Comment:
Just for the sake of clarity since reading this condition can be difficult for new comers:

```javascript
// The 'this._hookMemo[hookIndex] &&' safety check is not necessary since your `memo` can't be undefined
const isMemoizedHook = this._hookMemo[hookIndex] === memo
// pas sur de comprendre la derniere mais:
const isSame = this._hookMemo[hookIndex].find(runningMemo => runningMemo === memo)

if (!isMemoizedHook || !isSame) {}
```

Reply:
I will improve that :)